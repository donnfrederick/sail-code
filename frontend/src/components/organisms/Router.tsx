import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
