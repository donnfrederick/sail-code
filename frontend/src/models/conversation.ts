import { Questionnaire } from 'models/evaluations'
import { UserProfile as User } from 'models/userProfile'

export type TimeSelectType = 'from' | 'zone'

export type ConversationMode = 'soundOnly' | 'video'

export type ReservationType = 'anyone' | 'request'

export interface Response {
  channel_id: string
  chats: any[]
  id: number
  status: string
  start_at: string
  end_at: string
  created_at: string
  updated_at: string
  users: any[]
}

export interface ConversationRequestPostParams {
  reservable_conversation_id: number
  start_at: string
}

export interface Request {
  accepting_requests: number
  start_at: string
  end_at: string
}

export interface ConversationRequest {
  start_on?: string
  end_on?: string
  term?: 'week' | 'month'
  page?: number
}

export interface UserConversation {
  accepting_requests: boolean
  channel_id: string
  chats: any[]
  id: number
  status: string
  statuses: ConversationStatus[]
  start_at: string
  end_at: string
  evaluate: ConversationRate<Evaluate>[]
  memos: ConversationRate<Memo>[]
  reports: ConversationRate<Report>[]
  created_at: string
  updated_at: string
  with_self: boolean
}

export interface Chats {
  content: string
  conversation_id: number
  created_at: string
  user_picture_url: string
  id: number
  updated_at: string
  user_id: number
}

export interface Conversation {
  accepting_requests: boolean
  available: boolean
  channel_id: string
  chats: Chats[]
  id: number
  status: string
  statuses: ConversationStatus[]
  start_at: string
  end_at: string
  evaluate: ConversationRate<Evaluate>[]
  memos: ConversationRate<Memo>[]
  reports: ConversationRate<Report>[]
  created_at: string
  updated_at: string
  users: User[]
  conversation_requests: RequestConversationInConversation[]
}

export interface RequestConversation {
  conversation: Conversation
  end_at: string
  id: number
  start_at: string
  user: User
}

export interface RequestConversationInConversation {
  end_at: string
  id: number
  start_at: string
  user: User
}

export interface CancelledConversation {
  conversation_id: number
  start_at: string
  end_at: string
  created_at: string
  updated_at: string
  users: User[]
}

export interface Meta {
  start_on: string
  end_on: string
}

export interface ServerTime {
  current_time: string
  current_time_epoch: number
  end_at: string
  end_at_epoch: number
}

export interface ConversationResponse {
  data: Conversation[]
  meta: Meta
}

export interface RequestConversationResponse {
  data: RequestConversation[]
}

export interface Calendar {
  [date: string]: {
    is_enabled: boolean
    is_reserved: boolean
  }
}

export interface Topics {
  [key: string]: string
}

export type ConversationStatusValue = 'Ontime' | 'Late' | 'Absent'

export interface ConversationStatus {
  user_id: number
  status: ConversationStatusValue
}

export type ConversationRate<T> = { [P in keyof T]: T[P] } & { user_id: number }

interface Memo {
  memo: string
}
interface Evaluate {
  evaluate: Questionnaire
}

interface Report {
  report: ReportContentRequest
}
export type ReportReasonsUnion = 1 | 2 | 3 | 4 | 5

export interface ReportContentRequest {
  report_detail: string
  report_reasons: ReportReasonsUnion[]
}

export enum TopicEnumJa {
  '料理',
  '読書',
  'スポーツ',
  '歴史',
  '音楽',
  '芸術',
  '哲学',
  '旅行',
  '社会'
}

export enum TopicEnumEn {
  'Cooking',
  'Reading',
  'Sports',
  'History',
  'Music',
  'Art',
  'Philosophy',
  'Travel',
  'Society'
}
