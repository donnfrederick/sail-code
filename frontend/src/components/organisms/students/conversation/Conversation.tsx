import ConversationBalloon from 'components/atoms/ConversationBalloon'
import RoundImage from 'components/atoms/RoundImage'
import NetworkStatus from 'components/atoms/students/NetworkStatus'
import RemainingTime from 'components/atoms/students/RemainingTime'
import ConversationButtons from 'components/molecules/students/ConversationButtons'
import CloseConversationModal from 'components/organisms/students/modal_contents/close_conversation'
import withConversationId from 'hocs/withConversationId'
import * as ConversationModels from 'models/conversation'
import * as React from 'react'
import styled from 'styled-components'
import getPartner from 'utils/getPartner'
import resolvePath from 'utils/resolvePath'
import * as SoraClient from 'utils/SoraClient'
import Chat from 'components/molecules/Chat'

interface Props {
  bandwidth: number
  chats: ConversationModels.Chats[]
  closeSoon: boolean
  conversation: ConversationModels.Conversation | null
  conversationMode: ConversationModels.ConversationMode
  currentTopic: string
  isConnecting: boolean
  myId: number
  remainingTime: string
  topics: ConversationModels.Topics
  youAreLate: boolean
  openChat(): void
  openModal(): void
  setModalContents(contents: JSX.Element): void
}

export default withConversationId((props: Props) => {
  const {
    bandwidth,
    chats,
    closeSoon,
    conversation,
    conversationMode,
    // currentTopic,
    isConnecting,
    myId,
    remainingTime,
    // topics,
    youAreLate,
    openModal,
    setModalContents
  } = props

  if (conversation) {
    SoraClient.init(conversation.channel_id)
    SoraClient.connect()
  }

  const [isChatOpen, setIsChatOpen] = React.useState(false)
  const partner = conversation ? getPartner(conversation, myId) : null

  return (
    <Container>
      <RemoteVideoContainer>
        <RemoteVideo id="remote-video" autoPlay={true} playsInline={true} />
      </RemoteVideoContainer>
      <LocalVideoContainer>
        <LocalVideo id="local-video" autoPlay={true} playsInline={true} />
      </LocalVideoContainer>
      <NetworkStatus
        status={bandwidth > 300 ? 'good' : bandwidth > 100 ? 'normal' : 'bad'}
      />
      <RemainingTime remainingTime={remainingTime} />
      {conversationMode === 'soundOnly' ? (
        <Partner>
          <RoundImage
            src={
              partner
                ? partner.picture_url
                : resolvePath.image('common/user.png')
            }
            size={200}
            badge={partner ? (partner.highly_reliable ? 'elder' : null) : null}
            code={partner ? partner.country_code : ''}
          />
          <PartnerName>{partner ? partner.name : null}</PartnerName>
        </Partner>
      ) : null}
      {/* {isChatOpen ? <Topics topics={topics} /> : null}
      {!isChatOpen && currentTopic ? (
        <ConversationBalloon
          text={translateTopicToEnglish(currentTopic)}
          isTopic={true}
        />
      ) : null} */}
      {closeSoon ? (
        <ConversationBalloon text="It is time to say goodbye" />
      ) : null}
      {youAreLate ? (
        <ConversationBalloon text="You are late！" color="#ff676e" />
      ) : null}
      {isConnecting ? <Connecting>{'Connecting...'}</Connecting> : null}
      <ConversationButtons
        conversationMode={conversationMode}
        isChatOpen={isChatOpen}
        openChat={() => setIsChatOpen(!isChatOpen)}
        closeEvent={() => {
          setModalContents(<CloseConversationModal />)
          openModal()
        }}
      />
      <Canvas id="check-stream" />
      {conversation &&
        isChatOpen && (
          <ChatContainer>
            <Chat
              chats={chats}
              conversationId={conversation.id}
              openChat={() => setIsChatOpen(!isChatOpen)}
            />
          </ChatContainer>
        )}
    </Container>
  )
})

const Container = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  background-color: #000;
`

const LocalVideoContainer = styled.div`
  position: absolute;
  top: 32px;
  right: 32px;
  width: 148px;
  height: 238px;
  border-radius: 8px;
  overflow: hidden;
`

const LocalVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const RemoteVideoContainer = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 80%;
  overflow: hidden;
`

const RemoteVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const Partner = styled.div`
  position: absolute;
  top: 410px;
  left: 0;
  width: 100%;
  text-align: center;
`

const PartnerName = styled.div`
  margin-top: 32px;
  font-size: 48px;
  font-weight: 500;
  letter-spacing: 0.5px;
  text-align: center;
  color: #ffffff;
`

const Connecting = styled.div`
  position: absolute;
  top: 280px;
  right: 0;
  left: 0;
  width: 200px;
  height: 32px;
  margin: auto;
  font-size: 32px;
  color: #fff;
`

const Canvas = styled.canvas`
  display: none;
  width: 100%;
  height: 100%;
`

const ChatContainer = styled.div`
  position: absolute;
  background-color: #fff;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 50%;
  overflow: hidden;
`
// visibility: ${isChatOpen ? 'visible' : 'hidden'};
