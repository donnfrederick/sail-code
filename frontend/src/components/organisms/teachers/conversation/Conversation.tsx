import ConversationBalloon from 'components/atoms/ConversationBalloon'
import RoundImage from 'components/atoms/RoundImage'
import NetworkStatus from 'components/atoms/teachers/NetworkStatus'
import RemainingTime from 'components/atoms/teachers/RemainingTime'
import ConversationButtons from 'components/molecules/teachers/ConversationButtons'
// import Topics from 'components/molecules/Topics'
import CloseConversationModal from 'components/organisms/teachers/modal_contents/close_conversation'
import withConversationId from 'hocs/withConversationId'
import * as ConversationModels from 'models/conversation'
import * as React from 'react'
import styled from 'styled-components'
import getPartner from 'utils/getPartner'
import resolvePath from 'utils/resolvePath'
import * as SoraClient from 'utils/SoraClient'
import Chat from 'components/molecules/Chat'
// import { translateTopicToJapanese } from 'utils/translateTopic'

interface Props {
  bandwidth: number
  chats: ConversationModels.Chats[]
  closeSoon: boolean
  conversation: ConversationModels.Conversation | null
  conversationMode: ConversationModels.ConversationMode
  currentTopic: string
  isConnecting: boolean
  myId: number
  remainingTime: string
  topics: ConversationModels.Topics
  youAreLate: boolean
  closeModal(): void
  openModal(): void
  setModalContents(contents: JSX.Element): void
}

export default withConversationId((props: Props) => {
  const {
    bandwidth,
    chats,
    closeSoon,
    conversation,
    conversationMode,
    // currentTopic,
    isConnecting,
    myId,
    remainingTime,
    // topics,
    youAreLate,
    openModal,
    setModalContents
  } = props

  if (conversation) {
    SoraClient.init(conversation.channel_id)
    SoraClient.connect()
  }

  const [isChatOpen, setIsChatOpen] = React.useState(false)
  const partner = conversation ? getPartner(conversation, myId) : null

  // const openConversationCloseModal = () => {
  //   setModalContents(<CloseConversationModal />)
  //   openModal()
  // }

  return (
    <Container>
      <RemoteVideoContainer>
        <RemoteVideo id="remote-video" autoPlay={true} playsInline={true} />
      </RemoteVideoContainer>
      <LocalVideoContainer>
        <LocalVideo id="local-video" autoPlay={true} playsInline={true} />
      </LocalVideoContainer>
      <NetworkStatus
        status={bandwidth > 300 ? 'good' : bandwidth > 100 ? 'normal' : 'bad'}
      />
      <RemainingTime remainingTime={remainingTime} />
      {conversationMode === 'soundOnly' ? (
        <Partner>
          <RoundImage
            src={
              partner
                ? partner.picture_url
                : resolvePath.image('common/user.png')
            }
            size={200}
            code={partner ? partner.country_code : ''}
          />
          <PartnerName>{partner ? partner.name : null} さん</PartnerName>
        </Partner>
      ) : null}
      {/* {isTopicsOpen ? <Topics topics={topics} /> : null}
      {!isTopicsOpen && currentTopic ? (
        <ConversationBalloon
          text={translateTopicToJapanese(currentTopic)}
          isTopic={true}
        />
      ) : null} */}
      {closeSoon ? (
        <ConversationBalloon text="そろそろお別れの時間です" />
      ) : null}
      {youAreLate ? (
        <ConversationBalloon text="遅刻ですよ！" color="#ff676e" />
      ) : null}
      {isConnecting ? <Connecting>{'接続中...'}</Connecting> : null}
      <ConversationButtons
        conversationMode={conversationMode}
        isChatOpen={isChatOpen}
        openChat={() => setIsChatOpen(!isChatOpen)}
        closeEvent={() => {
          setModalContents(<CloseConversationModal />)
          openModal()
        }}
      />
      <Canvas id="check-stream" />
      {conversation &&
        isChatOpen && (
          <ChatContainer>
            <Chat
              chats={chats}
              conversationId={conversation.id}
              openChat={() => setIsChatOpen(!isChatOpen)}
            />
          </ChatContainer>
        )}
    </Container>
  )
})

const Container = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  background-color: #000;
`

const LocalVideoContainer = styled.div`
  position: absolute;
  top: 32px;
  right: 32px;
  width: 154px;
  height: 246px;
  overflow: hidden;
`

const LocalVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const RemoteVideoContainer = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  overflow: hidden;
`

const RemoteVideo = styled.video`
  position: absolute;
  top: 50%;
  left: 50%;
  height: 100%;
  transform: translate(-50%, -50%);
`

const Partner = styled.div`
  position: absolute;
  top: 410px;
  left: 0;
  width: 100%;
  text-align: center;
`

const PartnerName = styled.div`
  margin-top: 32px;
  font-size: 48px;
  font-weight: 500;
  letter-spacing: 0px;
  text-align: center;
  color: #ffffff;
`

const Connecting = styled.div`
  position: absolute;
  top: 280px;
  right: 0;
  left: 0;
  width: 200px;
  height: 40px;
  margin: auto;
  font-size: 40px;
  color: #fff;
`

const Canvas = styled.canvas`
  display: none;
  width: 100%;
  height: 100%;
`

const ChatContainer = styled.div`
  position: absolute;
  background-color: #fff;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 50%;
  overflow: hidden;
`
