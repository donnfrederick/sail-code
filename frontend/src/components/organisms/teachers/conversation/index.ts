import Conversation from 'components/organisms/teachers/conversation/Conversation'
import * as ConversationModels from 'models/conversation'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as ModalActions from 'reducers/modal'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  bandwidth: number
  chats: any[]
  closeSoon: boolean
  conversation: ConversationModels.Conversation | null
  conversationMode: ConversationModels.ConversationMode
  currentTopic: string
  isConnecting: boolean
  isTopicsOpen: boolean
  myId: number
  remainingTime: string
  topics: ConversationModels.Topics
  youAreLate: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    bandwidth: state.rootReducer.conversation.bandwidth,
    chats: state.rootReducer.conversation.chats,
    closeSoon: state.rootReducer.conversation.closeSoon,
    conversation: state.rootReducer.conversation.conversation,
    conversationMode: state.rootReducer.conversation.conversationMode,
    currentTopic: state.rootReducer.conversation.currentTopic,
    isConnecting: state.rootReducer.conversation.isConnecting,
    isTopicsOpen: state.rootReducer.conversation.isTopicsOpen,
    myId: state.rootReducer.teachers.me.id,
    remainingTime: state.rootReducer.conversation.remainingTime,
    topics: state.rootReducer.conversation.topics,
    youAreLate: state.rootReducer.conversation.youAreLate
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    closeModal: ModalActions.close,
    openModal: ModalActions.open,
    setModalContents: ModalActions.setContents
  }
)

export default enhancer(Conversation)
