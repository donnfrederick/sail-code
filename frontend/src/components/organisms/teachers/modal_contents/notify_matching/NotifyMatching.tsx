import RoundImage from 'components/atoms/RoundImage'
import Button from 'components/atoms/teachers/Button'
// import Stars from 'components/molecules/Stars'
import ModalTextContainer from 'components/molecules/teachers/ModalTextContainer'
import * as ConversationModels from 'models/conversation'
import * as React from 'react'
import styled from 'styled-components'
import getPartner from 'utils/getPartner'
import resolvePath from 'utils/resolvePath'

interface Props {
  conversation: ConversationModels.Conversation
  myId: number
  path: string
  close(): void
}

export default (props: Props) => {
  const { conversation, myId, path, close } = props

  const partner = getPartner(conversation, myId)

  return (
    <Container>
      <ModalTextContainer
        heading={'会話相手が見つかりました'}
        text={'詳細を確認してみましょう'}
      />
      <RoundImage
        src={partner ? partner.picture_url : ''}
        size={200}
        marginBottom={24}
        code={partner ? partner.country_code : ''}
      />
      <Name>{partner ? partner.name : ''}</Name>
      {/* <Stars
        score={partner ? partner.evaluate : 0}
        size={50}
        marginBottom={64}
      /> */}
      <ButtonContainer>
        <Button
          type="white"
          text="とじる"
          width={272}
          height={112}
          fontSize={40}
          onClick={() => close()}
        />
        <Button
          type="blue"
          text="詳細を見る"
          width={272}
          height={112}
          fontSize={40}
          link={resolvePath.page('teachers', path)}
          onClick={() => close()}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  height: 100%;
  text-align: center;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`

const Name = styled.div`
  margin-bottom: 40px;
  font-size: 40px;
  font-weight: 500;
  letter-spacing: 0px;
  text-align: center;
  color: #405766;

  &::after {
    margin-left: 8px;
    font-size: 32px;
    content: 'さん';
  }
`
