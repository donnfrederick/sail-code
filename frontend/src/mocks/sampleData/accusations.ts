import { conversation as sampleConversation } from 'mocks/sampleData/conversation'
import { me } from 'mocks/sampleData/me'
import * as AccusationsModels from 'models/accusations'

export const reasons: AccusationsModels.Reason[] = [
  {
    id: 1,
    name: 'reason 1'
  },
  {
    id: 2,
    name: 'reason 2'
  },
  {
    id: 3,
    name: 'reason 3'
  }
]

export const accusationsResponse: AccusationsModels.AccusationsResponse = {
  conversation: sampleConversation,
  from_user: me,
  to_user: me
}
