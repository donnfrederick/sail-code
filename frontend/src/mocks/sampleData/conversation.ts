import * as ConversationModels from 'models/conversation'

export const conversation: ConversationModels.Conversation = {
  accepting_requests: true,
  available: true,
  channel_id: '8e85f8e2-ebad-40f2-98be-b88efea45b53',
  chats: [],
  id: 26227,
  status: 'queued',
  statuses: [
    {
      user_id: 1,
      status: 'Ontime'
    },
    {
      user_id: 2,
      status: 'Ontime'
    }
  ],
  start_at: '2018-01-02T00:00:00+09:00',
  end_at: '2018-01-02T00:25:00+09:00',
  created_at: '2018-01-01T00:00:00+09:00',
  updated_at: '2018-01-01T00:00:00+09:00',
  evaluate: [
    {
      user_id: 1,
      evaluate: {
        fun: 1,
        ability: 1,
        time: 1,
        quality: []
      }
    },
    {
      user_id: 2,
      evaluate: {
        fun: 1,
        ability: 1,
        time: 1,
        quality: []
      }
    }
  ],
  memos: [
    {
      user_id: 1,
      memo: 'memo'
    },
    {
      user_id: 2,
      memo: 'memo'
    }
  ],
  reports: [
    {
      user_id: 1,
      report: {
        report_detail: 'report',
        report_reasons: [1, 2]
      }
    },
    {
      user_id: 2,
      report: {
        report_detail: 'report',
        report_reasons: [1, 2]
      }
    }
  ],
  users: [
    {
      absence: 0,
      id: 60331,
      username: 'teacher 29',
      name: '吉野 瑞葵',
      type: 'Teacher',
      sex: 2,
      picture_url: '',
      conversation_level: 5,
      level: 1,
      country: 'ja',
      country_code: 'ja',
      timezone: 'Asia/Tokyo',
      desired_condition: 2,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: null,
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'aho',
      email: ''
    },
    {
      absence: 0,
      id: 60332,
      username: 'student 29',
      name: '宮本 彩',
      type: 'Student',
      sex: 9,
      picture_url: '',
      conversation_level: 3,
      level: 3,
      country: 'TJ',
      country_code: 'TJ',
      timezone: 'Asia/Dushanbe',
      desired_condition: 1,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: 'member',
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'hage',
      email: ''
    }
  ],
  conversation_requests: []
}

export const cancelledConversation: ConversationModels.CancelledConversation = {
  conversation_id: 26227,
  start_at: '2018-01-02T00:00:00+09:00',
  end_at: '2018-01-02T00:25:00+09:00',
  created_at: '2018-01-01T00:00:00+09:00',
  updated_at: '2018-01-01T00:00:00+09:00',
  users: [
    {
      absence: 0,
      id: 60331,
      username: 'teacher 29',
      name: '吉野 瑞葵',
      type: 'Teacher',
      sex: 2,
      picture_url: '',
      level: 1,
      country: 'ja',
      country_code: 'ja',
      timezone: 'Asia/Tokyo',
      desired_condition: 2,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: null,
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'cancelled_aho',
      email: ''
    },
    {
      absence: 0,
      id: 60332,
      username: 'student 29',
      name: '宮本 彩',
      type: 'Student',
      sex: 9,
      picture_url: '',
      level: 3,
      country: 'TJ',
      country_code: 'TJ',
      timezone: 'Asia/Dushanbe',
      desired_condition: 1,
      evaluate: {
        1: 0,
        2: 0,
        3: 0,
        4: 0
      },
      location: '',
      lateness: 0,
      grade: 'member',
      highly_reliable: false,
      hobbies: [],
      payment_state: 'hidden',
      purposes: [],
      rated_conversation_level: 0,
      conversation_level: 0,
      is_blocked: false,
      is_favorite: false,
      conversations: [],
      introduce: 'cancelled_hage',
      email: ''
    }
  ]
}

export const requestConversationInConversation: ConversationModels.RequestConversationInConversation = {
  end_at: '2018-01-05T23:59:59+09:00',
  id: 1,
  start_at: '2018-01-02T00:00:00+09:00',
  user: {
    absence: 0,
    id: 60331,
    username: 'teacher 29',
    name: '吉野 瑞葵',
    type: 'Teacher',
    sex: 2,
    picture_url: '',
    conversation_level: 5,
    level: 1,
    country: 'ja',
    country_code: 'ja',
    timezone: 'Asia/Tokyo',
    desired_condition: 2,
    evaluate: {
      1: 0,
      2: 0,
      3: 0,
      4: 0
    },
    location: '',
    lateness: 0,
    grade: null,
    highly_reliable: false,
    hobbies: [],
    payment_state: 'hidden',
    purposes: [],
    rated_conversation_level: 0,
    is_blocked: false,
    is_favorite: false,
    conversations: [],
    introduce: 'hagehage',
    email: ''
  }
}

export const requestConversation = {
  ...requestConversationInConversation,
  conversation
}

export const conversations: ConversationModels.ConversationResponse = {
  data: [conversation],
  meta: {
    start_on: '2018-01-02T00:00:00+09:00',
    end_on: '2018-01-05T23:59:59+09:00'
  }
}

export const requestConversations: ConversationModels.RequestConversationResponse = {
  data: [requestConversation]
}

export const teachersRequestConversations: ConversationModels.Conversation[] = [
  conversation
]

export const calendar: ConversationModels.Calendar = {
  '1': {
    is_enabled: true,
    is_reserved: false
  },
  '2': {
    is_enabled: true,
    is_reserved: false
  },
  '3': {
    is_enabled: true,
    is_reserved: false
  },
  '4': {
    is_enabled: true,
    is_reserved: false
  },
  '5': {
    is_enabled: true,
    is_reserved: false
  },
  '6': {
    is_enabled: true,
    is_reserved: false
  },
  '7': {
    is_enabled: true,
    is_reserved: false
  },
  '8': {
    is_enabled: true,
    is_reserved: false
  },
  '9': {
    is_enabled: true,
    is_reserved: false
  },
  '10': {
    is_enabled: true,
    is_reserved: false
  },
  '11': {
    is_enabled: false,
    is_reserved: true
  },
  '12': {
    is_enabled: false,
    is_reserved: true
  },
  '13': {
    is_enabled: false,
    is_reserved: true
  },
  '14': {
    is_enabled: false,
    is_reserved: true
  },
  '15': {
    is_enabled: false,
    is_reserved: true
  },
  '16': {
    is_enabled: false,
    is_reserved: true
  },
  '17': {
    is_enabled: false,
    is_reserved: true
  },
  '18': {
    is_enabled: false,
    is_reserved: true
  },
  '19': {
    is_enabled: false,
    is_reserved: true
  },
  '20': {
    is_enabled: false,
    is_reserved: true
  },
  '21': {
    is_enabled: false,
    is_reserved: false
  },
  '22': {
    is_enabled: false,
    is_reserved: false
  },
  '23': {
    is_enabled: false,
    is_reserved: false
  },
  '24': {
    is_enabled: false,
    is_reserved: false
  },
  '25': {
    is_enabled: false,
    is_reserved: false
  },
  '26': {
    is_enabled: false,
    is_reserved: false
  },
  '27': {
    is_enabled: false,
    is_reserved: false
  },
  '28': {
    is_enabled: false,
    is_reserved: false
  },
  '29': {
    is_enabled: false,
    is_reserved: false
  },
  '30': {
    is_enabled: false,
    is_reserved: false
  },
  '31': {
    is_enabled: false,
    is_reserved: false
  }
}

export const topics = {
  Traveling: '旅行',
  Reading: '読書',
  Art: '芸術'
}
