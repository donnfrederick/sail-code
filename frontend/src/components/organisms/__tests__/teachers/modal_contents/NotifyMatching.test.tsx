import Router from 'components/organisms/Router'
import NotifyMatching from 'components/organisms/teachers/modal_contents/notify_matching'
import { conversation } from 'mocks/sampleData/conversation'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <NotifyMatching />', () => {
  const tree = renderer
    .create(
      <Router>
        <NotifyMatching
          conversation={conversation}
          path={'reservations/detail/1'}
        />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
