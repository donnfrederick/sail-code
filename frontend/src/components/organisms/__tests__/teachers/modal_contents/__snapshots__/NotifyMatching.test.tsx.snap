// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <NotifyMatching /> 1`] = `
<div
  className="sc-cSHVUG hsIUBe"
>
  <div
    className="sc-jzJRlG bkdbxf"
  >
    <div
      className="sc-jTzLTM llGFMw"
    >
      会話相手が見つかりました
    </div>
    <p
      className="sc-fjdhpX fXhuWY"
    >
      詳細を確認してみましょう
    </p>
  </div>
  <div
    className="sc-EHOje fbzZrP"
  >
    <img
      className="sc-gzVnrw QJFQI"
      height="200px"
      src=""
      srcSet=""
      width="200px"
    />
  </div>
  <div
    className="sc-chPdSV eqkTzc"
  >
    吉野 瑞葵
  </div>
  <div
    className="sc-kAzzGY iutItN"
  >
    <button
      className="sc-htoDjs iiCvpw"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      width={272}
    >
      <div
        className="sc-VigVT dXzNKq"
      >
        とじる
      </div>
    </button>
    <button
      className="sc-dnqmqq eXECE"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      width={272}
    >
      <a
        className="sc-gqjmRU klaAAy"
        href="/teachers/reservations/detail/1"
        onClick={[Function]}
      >
        詳細を見る
      </a>
    </button>
  </div>
</div>
`;
